java/browse-wfs-layers/src/main/java/com/esri/arcgisruntime/sample/browsewfslayers/MainActivity.java

```diff
@@ -0,0 +1,234 @@
+
+  private WfsLayerInfo mSelectedWfsLayerInfo;
+
+  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    mMapView = findViewById(R.id.mapView);
+    mAxisCheckbox = findViewById(R.id.axisCheckbox);
+    mLayersRecyclerView = findViewById(R.id.layersRecyclerView);
+    mLoadingView = findViewById(R.id.loadingView);
+
+    View bottomSheet = findViewById(R.id.bottomSheet);
+    mBottomSheetBehavior = BottomSheetBehavior.from(bottomSheet);
+    mBottomSheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+
+    mMapView.setOnTouchListener(new DefaultMapViewOnTouchListener(this, mMapView) {
+      @Override public boolean onTouch(View view, MotionEvent motionEvent) {
+        if (mBottomSheetBehavior.getState() == BottomSheetBehavior.STATE_EXPANDED) {
+          mBottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+          return true;
+        }
+        return super.onTouch(view, motionEvent);
+      }
+    });
+
+    mAxisCheckbox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+      if (mSelectedWfsLayerInfo != null) {
+        updateMap(mSelectedWfsLayerInfo);
+      }
+    });
+
+    ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+    mMapView.setMap(map);
+
+    // create WFS service
+    WfsService service = new WfsService(getString(R.string.wfs_service_url));
+    service.addDoneLoadingListener(() -> {
+      if (service.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+        logErrorToUser(getString(R.string.error_wfs_service_load_failure, service.getLoadError().getMessage()));
+      } else {
+        setupRecyclerView(service.getServiceInfo().getLayerInfos());
+      }
+    });
+    service.loadAsync();
+  }
+
+  @Override public void onItemSelected(WfsLayerInfo wfsLayerInfo) {
+    mSelectedWfsLayerInfo = wfsLayerInfo;
+    updateMap(wfsLayerInfo);
+  }
+
+  private void updateMap(WfsLayerInfo wfsLayerInfo) {
+    mBottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+    mLoadingView.setVisibility(View.VISIBLE);
+
+    // clear existing layer infos
+    mMapView.getMap().getOperationalLayers().clear();
+
+    // create feature table
+    WfsFeatureTable featureTable = new WfsFeatureTable(wfsLayerInfo);
+
+    featureTable.setAxisOrder(mAxisCheckbox.isChecked() ? OgcAxisOrder.SWAP : OgcAxisOrder.NO_SWAP);
+
+    // set the table's feature request mode
+    featureTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+    // populate the table
+    ListenableFuture<FeatureQueryResult> featureQueryResultFuture = featureTable
+        .populateFromServiceAsync(new QueryParameters(), false, null);
+
+    // run when the table has been populated
+    featureQueryResultFuture.addDoneListener(() -> {
+      // create a feature layer from the table
+      FeatureLayer featureLayer = new FeatureLayer(featureTable);
```

Comment:
This can be moved out of the done listener like the other one. So only keep set viewpoint to extent and loading visibility gone in the listener